### PR TITLE
Fix(SQL/SAI): dawnblade marksman distance

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/20347
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/20354

## SOURCE:

The changes have been validated through:
- [x] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. https://github.com/azerothcore/azerothcore-wotlk/issues/20347
2. https://www.wowhead.com/wotlk/npc=24979/dawnblade-marksman#abilities;mode:n25
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Rorschach91** [comment]
I tested the issue, but It seems that marksmans don't work as they should.

**Comment by Rorschach91** [comment]
Quantum, could you try to look Sunfury Archmages (npcs located in Netherstorm). 
Maybe there is the solution in their smartai.

**Comment by IntelligentQuantum** [comment]
```sql
DELETE FROM `smart_scripts` WHERE `entryorguid` = 24979 AND `source_type` = 0;

INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(24979, 0, 0, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 35932, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Immolation Arrow every second'),
(24979, 0, 1, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 6660, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Shoot every second'),
(24979, 0, 2, 0, 9, 0, 100, 0, 0, 10, 0, 0, 0, 0, 21, 1, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Melee Range - Enable Movement'),
(24979, 0, 3, 0, 9, 0, 100, 0, 0, 10, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Melee Range - Start Melee Attacks'),
(24979, 0, 4, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 21, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Beyond Melee Range - Stop Movement'),
(24979, 0, 5, 0, 0, 0, 100, 0, 0, 0, 10000, 10000, 0, 0, 11, 35935, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - In Combat - Cast Immolation every 10 seconds');

```

@Rorschach91 can you try this

**Comment by Rorschach91** [comment]
> ```sql
> DELETE FROM `smart_scripts` WHERE `entryorguid` = 24979 AND `source_type` = 0;
> 
> INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
> (24979, 0, 0, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 35932, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Immolation Arrow every second'),
> (24979, 0, 1, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 6660, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Shoot every second'),
> (24979, 0, 2, 0, 9, 0, 100, 0, 0, 10, 0, 0, 0, 0, 21, 1, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Melee Range - Enable Movement'),
> (24979, 0, 3, 0, 9, 0, 100, 0, 0, 10, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Melee Range - Start Melee Attacks'),
> (24979, 0, 4, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 21, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Beyond Melee Range - Stop Movement'),
> (24979, 0, 5, 0, 0, 0, 100, 0, 0, 0, 10000, 10000, 0, 0, 11, 35935, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - In Combat - Cast Immolation every 10 seconds');
> ```
> 
> @Rorschach91 can you try this

Seems that it works. You can remove the zone_isle_of_queldanas.cpp from this pr and modify the sql file updating it with this new DB init.

**Review by Grimdhex — APPROVED** [review]

**Comment by Rorschach91** [comment]
@Grimdhex wait please. There are some abilities that shouldn't be in the script.

File: data/sql/updates/pending_db_world/rev_1731487213354414876.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 24979 AND `source_type` = 0;
+
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(24979, 0, 0, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 35932, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Immolation Arrow every second'),
+(24979, 0, 1, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 6660, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Shoot every second'),
+(24979, 0, 2, 0, 9, 0, 100, 0, 0, 10, 0, 0, 0, 0, 21, 1, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Melee Range - Enable Movement'),
```
Comment (sogladev) [inline data/sql/updates/pending_db_world/rev_1731487213354414876.sql:6]: No need to manually enable/disable

example hunter like smartAI: compare with 2345 Dun garok mountaineer

File: data/sql/updates/pending_db_world/rev_1731487213354414876.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 24979 AND `source_type` = 0;
+
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(24979, 0, 0, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 35932, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Immolation Arrow every second'),
+(24979, 0, 1, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 6660, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Shoot every second'),
```
Comment (sogladev) [inline data/sql/updates/pending_db_world/rev_1731487213354414876.sql:5]: Use CastFlags 64, prevent combat movement on cast

example hunter like smartAI: compare with 2345 Dun garok mountaineer

Comment (sogladev) [reply data/sql/updates/pending_db_world/rev_1731487213354414876.sql:5]: No need to specify 9 RANGE the spells have min and max spell ranges
```
spellworks ID - 35932 Immolation Arrow
SpellRange: (Id 10) "Long Range":
    MinRange = 10, MinRangeFriendly = 10
    MaxRange = 40, MaxRangeFriendly = 40
```

File: data/sql/updates/pending_db_world/rev_1731487213354414876.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 24979 AND `source_type` = 0;
+
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(24979, 0, 0, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 35932, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Immolation Arrow every second'),
```
Comment (sogladev) [inline data/sql/updates/pending_db_world/rev_1731487213354414876.sql:4]: style: You can generate comments with Keira3 so comments are consistent

File: data/sql/updates/pending_db_world/rev_1731487213354414876.sql [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+(24979, 0, 1, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 6660, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Shoot every second'),
+(24979, 0, 2, 0, 9, 0, 100, 0, 0, 10, 0, 0, 0, 0, 21, 1, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Melee Range - Enable Movement'),
+(24979, 0, 3, 0, 9, 0, 100, 0, 0, 10, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Melee Range - Start Melee Attacks'),
+(24979, 0, 4, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 21, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Beyond Melee Range - Stop Movement'),
+(24979, 0, 5, 0, 0, 0, 100, 0, 0, 0, 10000, 10000, 0, 0, 11, 35935, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - In Combat - Cast Immolation every 10 seconds');
```
[the remark this answers is not in the document]
Comment (IntelligentQuantum) [reply data/sql/updates/pending_db_world/rev_1731487213354414876.sql:9]: wowhead and warmane.com

[the remark this answers is not in the document]
Comment (sogladev) [reply data/sql/updates/pending_db_world/rev_1731487213354414876.sql:9]: ID - 35932 Immolation Arrow has as
Trigger spell (35935) Immolation.

no need to manually cast it

File: data/sql/updates/pending_db_world/rev_1731487213354414876.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+(24979, 0, 0, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 35932, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Immolation Arrow every second'),
+(24979, 0, 1, 0, 9, 0, 100, 0, 1000, 1000, 1000, 1000, 10, 70, 11, 6660, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Range Check 10-70 yards - Cast Shoot every second'),
+(24979, 0, 2, 0, 9, 0, 100, 0, 0, 10, 0, 0, 0, 0, 21, 1, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Melee Range - Enable Movement'),
+(24979, 0, 3, 0, 9, 0, 100, 0, 0, 10, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Melee Range - Start Melee Attacks'),
```
Comment (sogladev) [inline data/sql/updates/pending_db_world/rev_1731487213354414876.sql:7]: This should not be needed

**Comment by sogladev** [comment]
I can't confirm the spells but before changes flaming arrow was used to target the target practice creature (5202), this looks to be missing atm. Immolation arrow is the one that adds the dot and is cast on players?

File: data/sql/updates/pending_db_world/rev_1731487213354414876.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rorschach91) [reply data/sql/updates/pending_db_world/rev_1731487213354414876.sql:9]: > wowhead and warmane.com

As I said in the past, WoWhead is not 100% reliable. Warmane is a retro-server so it is less reliable then wowhead.
In any case, Flaming Arrow is a spell that the npc casts only out of combat and only on the target practice.
So, the original smartai must be let as it is (event 0 and 1).

You should only add the shoot ability in the smartai, with the flag 64 in it (as sogladev said).

**Comment by Rorschach91** [comment]
> I can't confirm the spells but before changes flaming arrow was used to target the target practice creature (5202), this looks to be missing atm. Immolation arrow is the one that adds the dot and is cast on players?

As far as I know, this npc has these abilities on the official servers:
Flamming Arrow: this ability is used only on the target practice (it is not used in combat).
Shoot: Only used in combat.
Standard melee attack: Only used in combat.

This is a video that I recorded during my tests in cataclysm classic ptr: [youtube](https://www.youtube.com/watch?v=WMSF5Itplto)

**Comment by Rorschach91** [comment]
This is my full-query from Keira3. 
It should work in this way.

@IntelligentQuantum @sogladev 

```sql
UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 24979;

DELETE FROM `smart_scripts` WHERE (`source_type` = 0 AND `entryorguid` = 24979);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(24979, 0, 0, 0, 10, 0, 100, 0, 0, 70, 7000, 13000, 0, 0, 11, 45101, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Within 0-70 Range Out of Combat LoS - Cast \'Flaming Arrow\''),
(24979, 0, 1, 0, 1, 0, 100, 0, 3000, 3000, 5000, 5000, 0, 0, 11, 45101, 0, 0, 0, 0, 0, 19, 5202, 26, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Out of Combat - Cast \'Flaming Arrow\''),
(24979, 0, 2, 0, 0, 0, 100, 0, 1000, 1000, 2000, 2000, 0, 0, 11, 6660, 64, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - In Combat - Cast \'Shoot\'');

**Comment by IntelligentQuantum** [comment]
> This is my full-query from Keira3. It should work in this way.
> 
> @IntelligentQuantum @sogladev
> 
> ```sql
> UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 24979;
> 
> DELETE FROM `smart_scripts` WHERE (`source_type` = 0 AND `entryorguid` = 24979);
> INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
> (24979, 0, 0, 0, 10, 0, 100, 0, 0, 70, 7000, 13000, 0, 0, 11, 45101, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Within 0-70 Range Out of Combat LoS - Cast \'Flaming Arrow\''),
> (24979, 0, 1, 0, 1, 0, 100, 0, 3000, 3000, 5000, 5000, 0, 0, 11, 45101, 0, 0, 0, 0, 0, 19, 5202, 26, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Out of Combat - Cast \'Flaming Arrow\''),
> (24979, 0, 2, 0, 0, 0, 100, 0, 1000, 1000, 2000, 2000, 0, 0, 11, 6660, 64, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - In Combat - Cast \'Shoot\'');
> ```

I'll check it at night thank you

**Comment by Rorschach91** [comment]
> I'll check it at night thank you

No problem dude. We are to fix and to learn how to fix well :)

**Comment by Rorschach91** [comment]
> I can't confirm the spells but before changes flaming arrow was used to target the target practice creature (5202), this looks to be missing atm. Immolation arrow is the one that adds the dot and is cast on players?

Ehm...talking abount flamming arrow, there is another issue that regard it: https://github.com/azerothcore/azerothcore-wotlk/issues/20354

But this is not the PR when we should talk it. This PR shoul only fix the distance combat issue.

**Comment by sogladev** [comment]
> This is my full-query from Keira3. It should work in this way.
> 
> @IntelligentQuantum @sogladev
> 
> ```sql
> UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 24979;
> 
> DELETE FROM `smart_scripts` WHERE (`source_type` = 0 AND `entryorguid` = 24979);
> INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
> (24979, 0, 0, 0, 10, 0, 100, 0, 0, 70, 7000, 13000, 0, 0, 11, 45101, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Within 0-70 Range Out of Combat LoS - Cast \'Flaming Arrow\''),
> (24979, 0, 1, 0, 1, 0, 100, 0, 3000, 3000, 5000, 5000, 0, 0, 11, 45101, 0, 0, 0, 0, 0, 19, 5202, 26, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Out of Combat - Cast \'Flaming Arrow\''),
> (24979, 0, 2, 0, 0, 0, 100, 0, 1000, 1000, 2000, 2000, 0, 0, 11, 6660, 64, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - In Combat - Cast \'Shoot\'');
> ```
Looks better, but

1. id 0: this causes the creature to target players up to 70yd and shoot a flaming arrow :laughing: I removed this entry
2. remove initial IC timer for shoot. Stops mob from moving forward on aggro. 
```
DELETE FROM `smart_scripts` WHERE (`entryorguid` = 24979) AND (`source_type` = 0) AND (`id` IN (1, 2));
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(24979, 0, 1, 0, 0, 0, 100, 0, 0, 0, 2000, 2000, 0, 0, 11, 6660, 64, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - In Combat - Cast \'Shoot\''),
(24979, 0, 2, 0, 4, 0, 100, 0, 0, 0, 0, 0, 0, 0, 92, 0, 45101, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - On Aggro - Interrupt Spell \'Flaming Arrow\'');

```
I feel it should stop casting flaming arrow (3 sec cast) and attack the player. No source on this, unlucky timing on the ptr video. Could check for this on PTR. 

Also 35932 Immolation Arrow is not added. I can't confirm if this is casts it at players. I looked at some "daily isle" footage and didn't see it. It can be checked on PTR as well.

Initia

**Comment by Rorschach91** [comment]
> Looks better, but
> 
> 1. id 0: this causes the creature to target players up to 70yd and shoot a flaming arrow 😆 I removed this entry
> 2. remove initial IC timer for shoot. Stops mob from moving forward on aggro.
> 
> I feel it should stop casting flaming arrow (3 sec cast) and attack the player. No source on this, unlucky timing on the ptr video. Could check for this on PTR.
> 
> Also, 35932 Immolation Arrow is not added. I can't confirm if this is cast at players. I looked at some "Daily Isle" footage and didn't see it. It can be checked on PTR as well.

DB updated. It seems that also this [report ](https://github.com/azerothcore/azerothcore-wotlk/issues/20354)has been fixed now.

**Comment by IntelligentQuantum** [comment]
> ```sql
> DELETE FROM `smart_scripts` WHERE (`entryorguid` = 24979) AND (`source_type` = 0) AND (`id` IN (0, 1));
> INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
> (24979, 0, 0, 0, 1, 0, 100, 0, 3000, 3000, 5000, 5000, 0, 0, 11, 45101, 0, 0, 0, 0, 0, 19, 5202, 26, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - Out of Combat - Cast \'Flaming Arrow\''),
> (24979, 0, 1, 0, 0, 0, 100, 0, 0, 0, 2000, 2000, 0, 0, 11, 6660, 64, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - In Combat - Cast \'Shoot\'');
> ```

it works fine

**Review by Rorschach91 — APPROVED** [review]

**Comment by sogladev** [comment]
@Rorschach91 No need to add the `UPDATE` statement  in 9c37616d62846161feb0d9e19cb292ce75994c8c The creature already has SmartAI is set

**Comment by Rorschach91** [comment]
> @Rorschach91 No need to add the `UPDATE` statement in [9c37616](https://github.com/azerothcore/azerothcore-wotlk/commit/9c37616d62846161feb0d9e19cb292ce75994c8c) The creature already has SmartAI is set

Removed. Let's see if the checks goes crazy another time.

**Comment by sogladev** [comment]
Formatting looks fine now, think CI requires the first line empty to add the date

> I feel it should stop casting flaming arrow (3 sec cast) and attack the player. No source on this, unlucky timing on the ptr video. Could check for this on PTR.

tested, figured I'd record this^

without aggro on interrupt `(24979, 0, 3, 0, 4, 0, 100, 0, 0, 0, 0, 0, 0, 0, 92, 0, 45101, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - On Aggro - Interrupt Spell \'Flaming Arrow\'');`

https://github.com/user-attachments/assets/aaf8e19b-1396-4192-a53c-e4ee33c92e18



https://github.com/user-attachments/assets/336b0155-d663-490f-9cb5-cea6dbbb4081

**Comment by Rorschach91** [comment]
> Formatting looks fine now, think CI requires the first line empty to add the date
> 
> > I feel it should stop casting flaming arrow (3 sec cast) and attack the player. No source on this, unlucky timing on the ptr video. Could check for this on PTR.
> 
> tested, figured I'd record this^
> 
> without aggro on interrupt `(24979, 0, 3, 0, 4, 0, 100, 0, 0, 0, 0, 0, 0, 0, 92, 0, 45101, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Dawnblade Marksman - On Aggro - Interrupt Spell \'Flaming Arrow\'');`
> 
>  dawnblade_marksman.mp4 
>  dawnblade_marksman_interrupt.mp4

Do you suggest to add this line on the DB import?

**Pull request merged**